File: Elin/Code/get_TreeHeight.py
#!/usr/bin/env py

"""Script that calculates tree heights given a file containing distance of each tree
    from its base and angle to its top. Outputs to csv called INPUT_FILE_NAME_treeheights_py.csv"""

__appname__ = "get_TreeHeight.py"
__author__ = "Elin Falla, ef16@ic.ac.uk"  # change - add names
__version__ = "0.0.1"

# change - import csv not needed

# Imports # #change
import numpy as np
import pandas as pd
import re
import sys

# Loading data from command line input

string = sys.argv[1]

# Remove relative path
string = re.sub(r'^.*/', '', string)
# Remove file extension
string = string.split(".")[0]

#change, removed commented code here
df = pd.read_csv("../Data/trees.csv")

# Function

def TreeHeight(degrees, distance):
    """Calculates the height of a tree given distance of each tree
        from its base and angle to its top, using the trigonometric formula""" # change - add docstring

    radians = np.radians(degrees)
    height = distance * np.tan(radians)
    return height

example = TreeHeight(37, 40)

#change - why is this printed?
print(f"The height of a tree with an angle of 37 degrees at distance 40m is {example}")

# Assigning the output of the function to a column

df["Tree.Height.m"] = TreeHeight(df["Angle.degrees"], df["Distance.m"]) #### Not working

# Creating a csv output

df.to_csv(f"../Results/{string}_treeheights_py.csv", index = False) # change filename

# Message to state complete

print("get_Treeheight.py complete") # change message

File: Elin/Code/get_TreeHeight.R
#!/usr/bin/env R

# This function calculates heights of trees given distance of each tree
# from its base and angle to its top, using the trigonometric formula
# 
# height = distance * tan(radians)
#
# ARGUMENTS
# degrees: The angle of elevation of tree
# distance: The distance from base of tree (e.g. meters)
#
# OUTPUT
# The heights of the tree, same units as "distance"

# Loading data from command line input

args <- commandArgs(trailingOnly = TRUE)

string <- args[1]
# Remove relative path
string <- gsub(".*/","",string)
# Remove file extension
string <- tools::file_path_sans_ext(string)


tree_data <- read.csv(args[1])

# Function

TreeHeight <- function(degrees, distance) {
    radians <- degrees * pi / 180
    height <- distance * tan(radians)
    
    return(height)
}

#change: added t to height - WHY IS THIS HERE?
print(paste("The height of a tree with an angle of 37 degrees at distance 40m is", as.character(TreeHeight(37,40))))

# Assigning the output of the function to a column

tree_data$Tree.Height.m <- TreeHeight(tree_data$Angle.degrees, tree_data$Distance.m)

# Creating a csv output

write.csv(tree_data, paste("../Results/", string, "_treeheights.csv", sep = ""), row.names = FALSE) # change: _treeheights

# Message to state complete

print("get_Treeheight.R complete") # change